modules/Quiz/resources/IVQModule.js

```diff
@@ -360,22 +363,24 @@
              * @param {*} openQuestionText 
              */
             submitAnswer:function(questionNr,selectedAnswer,openQuestionText){
-                var _this = this,isAnswered;
+                var _this = this;
+                var defer = $.Deferred();
                 $.cpObject.cpArray[questionNr].selectedAnswer = selectedAnswer;
-                if ($.cpObject.cpArray[questionNr].isAnswerd) {
-                    isAnswered = true;
-                }
-                else{
-                    isAnswered = false;
-                    $.cpObject.cpArray[questionNr].isAnswerd = true;
-                }
-                _this.KIVQApi.addAnswer(isAnswered,_this.i2q(selectedAnswer),_this.kQuizUserEntryId,questionNr,function(data){
+                $.cpObject.cpArray[questionNr].submitRequested = true;
+                var submitCallTimeout = setTimeout(function(){
+                    defer.reject();
+                }, _this.apiRequestTimeout);
+                _this.KIVQApi.addAnswer($.cpObject.cpArray[questionNr].isAnswerd,_this.i2q(selectedAnswer),_this.kQuizUserEntryId,questionNr,function(data){
+                    clearTimeout(submitCallTimeout);
                     if (!_this.checkApiResponse('Add question err -->',data)){
-                        return false;
-                    }else {
+                        defer.reject();
```

Comment:
@semarche-kaltura 
If we fail - do we still want to hold the "submitRequested" flag as true? Will that not make bugs on other logic?